homeassistant/components/ecovacs/vacuum.py

```diff
@@ -79,69 +91,77 @@ def on_error(self, error):
         to change, that will come through as a separate on_status event
         """
         if error == "no_error":
-            self._error = None
+            self.error = None
         else:
-            self._error = error
+            self.error = error
 
         self.hass.bus.fire(
             "ecovacs_error", {"entity_id": self.entity_id, "error": error}
         )
         self.schedule_update_ha_state()
 
     @property
-    def unique_id(self) -> str:
-        """Return an unique ID."""
-        return self.device.vacuum.get("did")
-
-    @property
-    def is_on(self) -> bool:
-        """Return true if vacuum is currently cleaning."""
-        return self.device.is_cleaning
+    def state(self) -> str | None:
+        """Return the state of the vacuum cleaner."""
+        if self.error is not None:
+            return STATE_ERROR
 
-    @property
-    def is_charging(self) -> bool:
-        """Return true if vacuum is currently charging."""
-        return self.device.is_charging
+        if self.device.is_cleaning:
+            return STATE_CLEANING
 
-    @property
-    def status(self) -> str | None:
-        """Return the status of the vacuum cleaner."""
-        return self.device.vacuum_status
+        if self.device.is_charging:
+            return STATE_DOCKED
 
-    def return_to_base(self, **kwargs: Any) -> None:
-        """Set the vacuum cleaner to return to the dock."""
+        if self.device.vacuum_status == sucks.CLEAN_MODE_STOP:
+            return STATE_PAUSED
```

Comment:
Looking at other integrations and also the documentation, I think this should be STATE_IDLE
```suggestion
            return STATE_IDLE
```